### PR TITLE
Add annotations support and valid test cases

### PR DESCRIPTION
**Pull request by stevespringett**
This implements #166 

Signed-off-by: Steve Springett <steve@springett.us>

### CODE REVIEW

File: schema/bom-1.5.proto [hunk 1]
```diff
@@ -675,3 +677,29 @@ enum VulnerabilityAffectedStatus {
...
+  // An optional identifier which can be used to reference the annotation elsewhere in the BOM. Every bom-ref MUST be unique within the BOM.
+  optional string bom_ref = 1;
+  // The object in the BOM identified by its bom-ref. This is often a component or service, but may be any object type supporting bom-refs.
+  repeated string subjects = 2;
+  // The organization, person, component, or service which created the textual content of the annotation.
+  AnnotatorChoice annotator = 3;
+  // The date and time (timestamp) when the annotation was created.
+  google.protobuf.Timestamp timestamp = 4;
+  // The textual content of the annotation.
+  string text = 5;
```
Comment (DarthHater) [inline schema/bom-1.5.proto:704]: Is the exclusion of "signature" because Protocol Buffers don't really need it?

File: schema/bom-1.5.xsd [hunk 2]
```diff
@@ -2103,6 +2103,111 @@ limitations under the License.
...
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="timestamp" type="xs:dateTime" minOccurs="1" maxOccurs="1">
+                <xs:annotation>
+                    <xs:documentation>The date and time (timestamp) when the annotation was created.</xs:documentation>
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="text" type="xs:string" minOccurs="1" maxOccurs="1">
+                <xs:annotation>
+                    <xs:documentation>The person(s) or organization(s) that published the component</xs:documentation>
```
Comment (jkowalleck) [inline schema/bom-1.5.xsd:2184]: this documentation string seams wrong.

in the json schema it is a differnt one: "The textual content of the annotation"

**Review by jkowalleck — CHANGES_REQUESTED** [review]

File: schema/bom-1.5.proto (hunk 1 above)
Comment (coderpatros) [reply schema/bom-1.5.proto:704]: I think because there isn't, as far as I know, a good way to sign parts of a protobuf message

**Review by coderpatros — CHANGES_REQUESTED** [review]
Happy with this except for @jkowalleck's comment

File: schema/bom-1.5.proto (hunk 1 above)
Comment (stevespringett) [reply schema/bom-1.5.proto:704]: Correct. I'm not aware of any solution for signing in protobuf.

File: schema/bom-1.5.xsd (hunk 2 above)
Comment (stevespringett) [reply schema/bom-1.5.xsd]: Good catch. Corrected.

**Pull request merged**